debian/syncd.init

```diff
@@ -34,68 +51,47 @@ start_bcm()
 start_mlnx()
 {
     [ -e /dev/sxdevs/sxcdev ] || ( mkdir -p /dev/sxdevs && mknod /dev/sxdevs/sxcdev c 231 193 )
-}
 
-case "$1" in
-start)
-    [ -r /etc/machine.conf ] && . /etc/machine.conf
-
-    if [ "$onie_platform" == "x86_64-dell_s6000_s1220-r0" ]; then
-        start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/dell_s6000.profile "
+    # Read MAC address and align the last 6 bits.
+    MAC_ADDRESS=`ip link show eth0 | grep ether | awk '{print $2}'`
+    last_byte=`python -c "print '$MAC_ADDRESS'[-2:]"`
+    aligned_last_byte=`python -c "print format(int(int('$last_byte', 16) & 0b11000000), '02x')"`  # put mask and take away the 0x prefix
+    ALIGNED_MAC_ADDRESS=`python -c "print '$MAC_ADDRESS'[:-2] + '$aligned_last_byte'"`          # put aligned byte into the end of MAC
 
-        if [ ${ENABLE_SAITHRIFT} == 1 ]; then
-            DAEMON_ARGS+=" -r -m /etc/syncd.d/dell_s6000_port_map.ini"
-        fi
-
-    elif [ "$onie_platform" == "x86_64-dell_s6100_c2538-r0" ]; then
-        start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/dell_s6100.profile -N"
-
-    elif [ "$onie_platform" == "x86_64-dell_z9100_c2538-r0" ]; then
-        start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/dell_z9100.profile"
+    # Write MAC address into /tmp/profile file.
+    cat $HWSKU_DIR/sai.profile > /tmp/sai.profile
+    echo "DEVICE_MAC_ADDRESS=$ALIGNED_MAC_ADDRESS" >> /tmp/sai.profile
+}
 
-    elif [ "$aboot_platform" == "x86_64-arista_7050_qx32" ]; then
-        start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/arista_a7050_qx32.profile"
+start_cavium()
+{
+    export XP_ROOT=/usr/bin/
+}
 
-    elif [ "$onie_platform" == "x86_64-m3000-r0" ]; then
-        start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/m3000_r0.profile"
+case "$1" in
+start)
+    eval $(parse_yaml /etc/sonic/sonic_version.yml "sonic_")
 
-    elif [ "$onie_platform" == "x86_64-m3000-r1" ]; then
+    if [ $sonic_asic_type == "broadcom" ]; then
         start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/m3000_r1.profile"
-
-    elif [ "$onie_platform" == "x86_64-mlnx_x86-r5.0.1400" ] || [ "$onie_platform" == "x86_64-mlnx_msn2700-r0" ]; then
+        DAEMON_ARGS+=" -p $HWSKU_DIR/sai.profile "
+    elif [ $sonic_asic_type == "mellanox" ]; then
         start_mlnx
+        DAEMON_ARGS+=" -p /tmp/sai.profile "
```

Comment:
question (may not be related): why mellanox sai.profile in /tmp/ folder instead of $HWSKU_DIR/ folder